### ISSUE TITLE
Fix scoring code

### ISSUE THREAD

**Issue by jaeddy** [Sage-Bionetworks/metadata-automation-challenge#62]
Running the scoring locally, I got the following scores:

```
APOLLO-2: 4.56666666666667
Outcome-Predictors: 4.70588235294118
REMBRANDT: 4.5625
ROI-Masks: 4.8125
```

_Originally posted by @vpchung in https://github.com/Sage-Bionetworks/metadata-automation-challenge/issues/61#issuecomment-597211624_

**Comment by jaeddy** [comment]
@vpchung - here are some steps to inspect/troubleshoot the scoring code:

1. Load one of the gold standard JSON files:

```r
anno_data <- jsonlite::read_json("data/Annotated-APOLLO-2.json")
```

2. Pull out data for an individual column and result:

```r
anno_res_data <- jsonlite::read_json("/data/Annotated-APOLLO-2.json") %>% 
  get_column_data(45) %>% 
  get_result_data(1) %>%
  # for columns with long lists of `valueDomain` records, this will keep
  # only the first 6 (for easier testing)
  purrr::modify_depth(2, head)
```

3. Get the score for value domain coverage:

```r
source("R/scoring.R")
score_value_coverage(anno_res_data, anno_res_data)
```

If things are working correctly, this should return `1` (if not, it will return `0` or some other number less than `1`).

---

That said, I think I spotted the problem in `score_value_coverage()`:

This [block of code](https://github.com/Sage-Bionetworks/metadata-automation-challenge/blob/master/R/scoring.R#L135-139)...
```r
  if (nrow(anno_vd) & length(mismatch_rows)) {
    1 - (length(mismatch_rows) / nrow(anno_vd))
  } else {
    0
  }
```

... should be this:
```r
  if (nrow(anno_vd)) {
    if (length(mismatch_rows)) {
      1 - (length(mismatch_rows) / nrow(anno_vd))
    } else {
      1
    }
  } else {
    0
  }
```

With that, I get a perfect score as expected. I'll create a PR with the fix, but others are welcome to double-check the code. 🙂

**Comment by jaeddy** [comment]
Fixed in #63
